ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -628,19 +619,28 @@ private NamedBlobRecord run_get_v2(String accountName, String containerName, Str
   private Page<NamedBlobRecord> run_list_v2(String accountName, String containerName, String blobNamePrefix,
       String pageToken, short accountId, short containerId, Connection connection, Integer maxKeys) throws Exception {
     String query = "";
-    String queryStatement = blobNamePrefix == null ? LIST_ALL_QUERY_V2 : LIST_QUERY_V2;
+    String queryStatement = blobNamePrefix == null ? LIST_ALL_QUERY_V2 : LIST_NAMED_BLOBS_SQL;
     int maxKeysValue = maxKeys == null ? config.listMaxResults : maxKeys;
     try (PreparedStatement statement = connection.prepareStatement(queryStatement)) {
-      statement.setInt(1, accountId);
-      statement.setInt(2, containerId);
       if (blobNamePrefix == null) {
+        // list-all no prefix
+        statement.setInt(1, accountId);
```

Comment:
I'm a little bit confused here.
You didn't change the LIST_ALL_QUERY_V2, why you can update the statement settings?

Reply:
Didn't update statement settings for list all. Just moved it into a separate case because that has different number of parameters.

Reply:
But if the blobNamePrefix is null, the queryStatement would become LIST_ALL_QUERY_V2 right? Am I miss anything here?